src/Domain/BlockLogicalCoordinates.hpp

```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <ostream>
+#include <vector>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
+template <size_t VolumeDim, typename TargetFrame>
+class Domain;
+namespace PUP {
+class er;
+}  // namespace PUP
+
+/// Holds block logical coords for a single point.
```

Comment:
To me the most natural data structure is `std::vector<tnsr::I<DataVector, Dim, Frame::Logical>`, which is basically what `ElementId` uses. Why this struct? Also, just use a `std::pair`, I don't see the benefit of writing a new class here. If we wanted to strongly type anything, we should strongly type `BlockId`, then the `std::pair` naturally picks up the info with no extra class.

Reply:
See the long description, as I don't understand what your comment means especially "ElementId uses".  Also @wthrowe and I proposed the struct instead of the `std::pair` because named things makes it much easier to figure out what the `size_t`s and tensors are.

Reply:
`ElementId` related things use a hash table. For block ids they are just dense integer values so we can use a `vector` as the "map". I find it a bit odd to use the map approach in one case and this in the other. Does that help clarify what I meant?

I think since we will have this sort of need come up a lot more that it would make more sense to make a `BlockId` class and a `BlockVector` that just wraps a `std::vector`. This would likely also clean up a lot of the domain code @gsb76 has written where corners and block ids are integers. This would make ElementId and BlockId more similar (both strongly typed), and also make classes that store a block id easier to reason about, since they'll then store a `BlockId` instead of a `size_t`.

We probably want to have a struct template that is general for these id sort of things:
```
template <typename Id, typename DataType>
struct IdPair {
  Id id;
  DataType value;
};
```
(I'm not at all attached to the name `IdPair`)

Reply:
Nope, still confused; the `block_id`s related with each target point and therefore held in individual `BlockLogicalCoordHolder`s are not dense, not ordered, and will usually be small subset of the overall set of `BlockId`s , so a `BlockVector` would not be useful here.